# Exploratory Data Analysis (EDA)

## Objective

The objective of this notebook is to analyze the cleaned Google Play Store dataset, identify key business insights, understand user and app trends, and generate findings that will support SQL analysis, Power BI dashboard development, and business recommendations.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv("../data/cleaned/google_playstore_cleaned.csv")

In [3]:
df.shape

(2312944, 23)

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2312944 entries, 0 to 2312943
Data columns (total 23 columns):
 #   Column             Dtype  
---  ------             -----  
 0   App Name           str    
 1   App Id             str    
 2   Category           str    
 3   Rating             float64
 4   Rating Count       float64
 5   Installs           float64
 6   Minimum Installs   float64
 7   Maximum Installs   int64  
 8   Free               bool   
 9   Price              float64
 10  Currency           str    
 11  Size               float64
 12  Minimum Android    str    
 13  Developer Id       str    
 14  Developer Website  str    
 15  Developer Email    str    
 16  Released           str    
 17  Last Updated       str    
 18  Content Rating     str    
 19  Privacy Policy     str    
 20  Ad Supported       bool   
 21  In App Purchases   bool   
 22  Editors Choice     bool   
dtypes: bool(4), float64(6), int64(1), str(12)
memory usage: 344.1 MB


In [5]:
total_apps = len(df)
print(f"Total Apps: {total_apps:,}")

Total Apps: 2,312,944


In [6]:
total_categories = df["Category"].nunique()
print(f"Total Categories: {total_categories}")

Total Categories: 48


In [7]:
df["Free"].value_counts()

Free
True     2267876
False      45068
Name: count, dtype: int64

In [8]:
free_percentage = (df["Free"].mean()) * 100

print(f"Free Apps: {free_percentage:.2f}%")
print(f"Paid Apps: {100-free_percentage:.2f}%")

Free Apps: 98.05%
Paid Apps: 1.95%


In [9]:
df["Editors Choice"].value_counts()

Editors Choice
False    2312091
True         853
Name: count, dtype: int64

In [10]:
summary = pd.DataFrame({
    "Metric": [
        "Total Apps",
        "Total Categories",
        "Free Apps (%)",
        "Paid Apps (%)",
        "Editors' Choice Apps"
    ],
    "Value": [
        len(df),
        df["Category"].nunique(),
        round(df["Free"].mean()*100,2),
        round((1-df["Free"].mean())*100,2),
        df["Editors Choice"].sum()
    ]
})

summary

,Metric,Value
0,Total Apps,2312944.00
1,Total Categories,48.00
2,Free Apps (%),98.05
3,Paid Apps (%),1.95
4,Editors' Choice Apps,853.00


### Key Insights

- The dataset contains **2312944** applications across **48** categories.
- Free apps constitute the majority of the Google Play Store.
- Only a small fraction of applications receive the **Editors' Choice** badge, indicating its exclusivity.